src/CP/core/search/ilds.jl

```diff
@@ -2,19 +2,19 @@
 
 
 """
-initroot!(toCall::Stack{Function}, ::Type{DFSearch},model::CPModel, variableHeuristic::AbstractVariableSelection, valueSelection::ValueSelection, newConstraints=nothing)
+initroot!(toCall::Stack{Function}, ::ILDSearch, model::CPModel, variableHeuristic::AbstractVariableSelection, valueSelection::ValueSelection)
 
-generic function to instantiate the research based on a specific Strategy <: SearchStrategy. The max discrepancy correspond to the number or branchable variables at 
-the beginning of the search. Calls to expandIlds! with a decreasing discrepancy is stacked in the toCall Stack. 
+generic function to instantiate the research based on a specific Strategy <: SearchStrategy. The max discrepancy correspond to the number of branchable variables 
+at the beginning of the search. Calls to expandIlds! with a decreasing discrepancy is stacked in the toCall Stack. 
 """
-function initroot!(toCall::Stack{Function}, ::Type{ILDSearch}, model::CPModel, variableHeuristic::AbstractVariableSelection, valueSelection::ValueSelection)
+function initroot!(toCall::Stack{Function}, ::ILDSearch , model::CPModel, variableHeuristic::AbstractVariableSelection, valueSelection::ValueSelection)
     isboundedlist = [!isbound(v) for (k,v) in model.variables]
     @assert !isempty(isboundedlist) "initialisation failed : no declared variables"
     depth = sum(isboundedlist)
     
     # Note that toCall stack is a LIFO data structure, expandIlds with a discrepancy threshold of 0 will be the first one to execute (then with 1, 2, 3, etc.)
     for k in depth:-1:1
-        push!(toCall, (model) -> (expandIlds!(toCall,k,depth, nothing, model, variableHeuristic, valueSelection)))
+        push!(toCall, (model) -> (restart_search!(model); expandIlds!(toCall,k,depth, nothing, model, variableHeuristic, valueSelection)))
```

Comment:
The call to `restart_search!` reset some metrics and ensure that we are back at the 
```suggestion
        push!(toCall, (model) -> (restart_search!(model); expandIlds!(toCall,k,depth, nothing, model, variableHeuristic, valueSelection)))top of the tree.
```